var/spack/repos/builtin/packages/mmg/package.py

```diff
@@ -72,6 +73,12 @@ def cmake_args(self):
             self.define("LIBMMGS_STATIC", not shared_active),
             self.define("LIBMMG_STATIC", not shared_active),
         ]
+        # Add conditional flag for MMG_INSTALL_PRIVATE_HEADERS
+        if self.spec.satisfies("@5.7.0:"):
+            args.append(
+                self.define("MMG_INSTALL_PRIVATE_HEADERS", self.spec.satisfies("+private_headers"))
+            )
```

Comment:
If this option only applies for versions `@5.7:`, then you could pass that as a `when` argument to the variant then not bother with the check here.